File: algorithm_introduction/ss32_string_matching/python/naive_string_matcher.py
def naive_string_matcher(T: str, P: str):
    n = len(T)
    m = len(P)

    for i in range(n - m + 1):
        matched = True
        for j in range(m):
            if P[j] != T[i + j]:
                matched = False
                break
        if matched:
            print('pattern occurs with s = %d' % i)


T = "abcabaabcabac"
P = "abaa"

naive_string_matcher(T, P)

File: algorithm_introduction/ss32_string_matching/cpp/naive_string_matcher.cpp
#include <stdio.h>
#include <string>

using namespace std;

void naive_string_matcher(string T, string P) {
  const char *T_arr = T.c_str();
  const char *P_arr = P.c_str();

  int n = (int)T.size();
  int m = (int)P.size();

  for (int i = 0; i <= n - m; ++i) {
    bool matched = true;
    for (int j = 0; j < m; ++j) {
      if (P_arr[j] != T_arr[i + j]) {
        matched = false;
        break;
      }
    }
    if (matched) {
      printf("pattern occurs with s = %d\n", i);
    }
  }
}

int main() {
  string T = "abcabaabcabac";
  string P = "abaa";

  naive_string_matcher(T, P);
}